### PR TITLE
Feature/#725 modify type adapter factory spec builder

### PR DESCRIPTION
**Pull request by gobbletsit**
https://app.productive.io/1-infinum/tasks/2547892

This PR contains changes made to simplify getTypeAdapter() usage for non retrofit users with passing either `Class<*>` or full `ParameterizedType` in case of lists.

### CODE REVIEW

**Review by stjepanbanek — CHANGES_REQUESTED** [review]
Good job! I do have 1 suggestion, so please check that out.

File: core/src/main/java/com/infinum/jsonapix/core/adapters/AdapterFactory.kt [hunk 1]
```diff
@@ -1,5 +1,10 @@
 package com.infinum.jsonapix.core.adapters
 
+import java.lang.reflect.ParameterizedType
+import kotlin.reflect.KClass
+
 interface AdapterFactory {
-    fun getAdapter(type: String): TypeAdapter<*>?
+    fun getAdapter(type: KClass<*>): TypeAdapter<*>?
+
+    fun getListAdapter(type: ParameterizedType): TypeAdapter<*>?
```
[the remark this answers is not in the document]
Comment (gobbletsit) [reply core/src/main/java/com/infinum/jsonapix/core/adapters/AdapterFactory.kt:9]: Are you sure? If so , then you would need to pass a KClass to it from `JsonXConverterFactory.responseBodyConverter` which is basically the same thing as initial implementation and as previous Sting implementation which doesn't solve the issue.

[the remark this answers is not in the document]
Comment (gobbletsit) [reply core/src/main/java/com/infinum/jsonapix/core/adapters/AdapterFactory.kt:9]: In that case, you will not be able to pull argument types qualified name from KClass like you can from ParameterizedType.

`(type.actualTypeArguments.first() as Class<*>).kotlin.qualifiedName`

[the remark this answers is not in the document]
Comment (gobbletsit) [reply core/src/main/java/com/infinum/jsonapix/core/adapters/AdapterFactory.kt:9]: We can only do it like this if you prefer but then we just moved it from SpecBuilder to here.

`adapterFactory.getListAdapter((type.actualTypeArguments.first() as Class<*>).kotlin)?.let {
                    JsonXResponseBodyConverter(it)
  }`

**Review by stjepanbanek — APPROVED** [review]

**Pull request merged**